### PR TITLE
[#4508]feat(iceberg) support load custom catalog as backend in the Iceberg REST service

### PR DESCRIPTION
**Pull request by theoryxu**
### What changes were proposed in this pull request?

support load custom catalog as backend

### Why are the changes needed?

improvement: #4508

### Does this PR introduce _any_ user-facing change?

1. add a property key

### How was this patch tested?

1. add UT
2. manual test

### CODE REVIEW

**Comment by theoryxu** [comment]
@jerryshao @FANNG1 please help to review, Thanks.

File: iceberg/iceberg-common/src/test/java/org/apache/gravitino/iceberg/common/utils/TestIcebergCatalogUtil.java [hunk 1]
```diff
@@ -78,4 +78,52 @@ void testLoadCatalog() {
...
+    config.clear();
+    config.put("catalog-backend-impl", "org.apache.iceberg.hive.HiveCatalog");
+    catalog =
+        IcebergCatalogUtil.loadCatalogBackend(
+            IcebergCatalogBackend.valueOf("CUSTOM"), new IcebergConfig(config));
+    Assertions.assertTrue(catalog instanceof HiveCatalog);
+
+    config.clear();
+    config.put(
+        "catalog-backend-impl", "org.apache.gravitino.iceberg.common.utils.CustomCatalogForTest");
```
Comment (FANNG1) [inline iceberg/iceberg-common/src/test/java/org/apache/gravitino/iceberg/common/utils/TestIcebergCatalogUtil.java:102]: seems use `org.apache.gravitino.iceberg.common.utils.CustomCatalogForTest` is enough, no need to test `memory` or `hive`?

**Comment by FANNG1** [comment]
Do you think is it necessary to support a custom catalog for the Gravitino Iceberg catalog not only for the Iceberg REST service?

File: iceberg/iceberg-common/src/test/java/org/apache/gravitino/iceberg/common/utils/TestIcebergCatalogUtil.java (hunk 1 above)
Comment (theoryxu) [reply iceberg/iceberg-common/src/test/java/org/apache/gravitino/iceberg/common/utils/TestIcebergCatalogUtil.java:102]: ok, I'll remove it

**Comment by theoryxu** [comment]
> Do you think is it necessary to support a custom catalog for the Gravitino Iceberg catalog not only for the Iceberg REST service?

I want to keep a PR as simple as possible. Maybe that part putting in another PR is better？

of course, if it's confusing, I will put them together.

**Comment by theoryxu** [comment]
> Do you think is it necessary to support a custom catalog for the Gravitino Iceberg catalog not only for the Iceberg REST service?

I planned to do it in a separate PR to keep this one simple, but to avoid confusion, I'll combine them.

**Comment by FANNG1** [comment]
> > Do you think is it necessary to support a custom catalog for the Gravitino Iceberg catalog not only for the Iceberg REST service?
> 
> I planned to do it in a separate PR to keep this one simple, but to avoid confusion, I'll combine them.

I prefer to split the PR to make it more clear, could you create corresponding issues to add custom catalog backend for Graviitno server?

File: docs/iceberg-rest-service.md [hunk 2]
```diff
@@ -162,6 +162,14 @@ If you have a JDBC Iceberg catalog prior, you must set `catalog-backend-name` to
 You must download the corresponding JDBC driver to the `iceberg-rest-server/libs` directory.
 :::
 
+#### Custom backend configuration
+| Configuration item                             | Description                                                                                                         | Default value    | Required | Since Version |
+|------------------------------------------------|---------------------------------------------------------------------------------------------------------------------|------------------|----------|---------------|
+| `gravitino.iceberg-rest.catalog-backend`       | The Catalog backend of the Gravitino Iceberg REST catalog service. Use the value **`custom`** for a Custom catalog. | `memory`         | Yes      | 0.7.0         |
```
Comment (FANNG1) [inline docs/iceberg-rest-service.md:168]: should be `0.2.0`

**Comment by FANNG1** [comment]
LGTM, except one comment

**Comment by theoryxu** [comment]
> > > Do you think is it necessary to support a custom catalog for the Gravitino Iceberg catalog not only for the Iceberg REST service?
> > 
> > 
> > I planned to do it in a separate PR to keep this one simple, but to avoid confusion, I'll combine them.
> 
> I prefer to split the PR to make it more clear, could you create corresponding issues to add custom catalog backend for Graviitno server?

👌，the corresponding issue is here: #4584

File: docs/iceberg-rest-service.md (hunk 2 above)
Comment (theoryxu) [reply docs/iceberg-rest-service.md]: fixed it

**Review by FANNG1 — APPROVED** [review]

**Comment by FANNG1** [comment]
@theoryxu , merged to main, thanks for your contribution

**Pull request merged**